src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -1117,7 +1117,10 @@ impl ThinPool {
                 meta_growth,
             );
 
-            (ext.is_ok(), ext)
+            match ext {
+                Ok(Sectors(0)) | Err(_) => (false, ext),
```

Comment:
Isn't this unreachable?

Reply:
@jbaublitz You might be misreading the diff.

Reply:
Ah I see, so this is an optimization to not save the metadata if it's not extended at all? I am failing to see a case where this will ever be reached. Unreachable might be the wrong term, because it could be reached, but `meta_growth` is guaranteed to be larger than 0. I don't see a case where this will be reached with the current implementation. I was originally thinking that `ext` could be 0 but I don't see a case where `meta_growth` would be non-zero but `ext` would be zero.

Reply:
Let me dig into do_extend and see if maybe it's rounding down. That could potentially cause the behavior you're seeing.

Reply:
I understand your argument. I think we can leave this in pending for now and work on simplifying it later.